addons/sale_purchase/tests/test_sale_purchase.py

```diff
@@ -305,3 +315,30 @@ def test_pol_custom_attribute(self):
         sale_order.action_confirm()
         pol = sale_order._get_purchase_orders().order_line
         self.assertEqual(pol.name, f"{self.service_purchase_1.display_name}\n\n{product_attribute.name}: {product_attribute_value.name}: {custom_value}")
+
+    def test_pol_analytic_distribution(self):
+        """Confirming SO, analytic account from SO should be set as Analytic Distribution in POL."""
+        self.sale_order_1.analytic_account_id, self.sale_order_2.analytic_account_id = self.test_analytic_account_1, self.test_analytic_account_1
+        self.sale_order_2.order_line.analytic_distribution = {str(self.sale_order_2.analytic_account_id.id): 100}
```

Comment:
Shouldn't you have 1 Sale Order with no `analytic_account_id` assigned to make sure that you're actually testing that the `sol.analytic_distribution` is correctly assigned? Or at least make this value different from the `sale_order.analytic_account_id`?